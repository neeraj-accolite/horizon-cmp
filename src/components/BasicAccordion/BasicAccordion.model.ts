export interface BasicAccordionProps {
  title: string;
  description: string;
  height?: string;
  width?: string;
}
