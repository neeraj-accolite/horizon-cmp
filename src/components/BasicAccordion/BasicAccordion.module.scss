.accordion {
    border-top: 1.5px solid #e0e0e0;
    border-bottom: 1.5px solid #e0e0e0;
}

.accordionTitle {
    padding: 16px;
    cursor: pointer;
    background-color: white;
    transition: background-color 0.3s ease;
    display: flex;
    justify-content: space-between;
    align-items: center;
    font-family: 'Inter', sans-serif;
    font-weight: 700;
    font-size: 16px;
    line-height: 24px;

    &:hover {
        background-color: #e0e0e0;
    }
}

.accordionContent {
    padding: 16px;
    display: none;
    font-family: 'Inter', sans-serif;
    font-weight: 400;
    font-size: 16px;
    line-height: 24px;
    margin-bottom: 12px;
}

.accordionContent.active {
    display: block;
}