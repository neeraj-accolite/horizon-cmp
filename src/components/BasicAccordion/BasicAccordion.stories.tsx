import React from 'react';
import { Meta, StoryFn } from '@storybook/react';
import { BasicAccordion } from './BasicAccordion';
import { BasicAccordionProps } from './BasicAccordion.model';

export default {
  title: 'Components/BasicAccordion',
  component: BasicAccordion,
} as Meta;

const Template: StoryFn<BasicAccordionProps> = (args) => (
  <BasicAccordion {...args} />
);

export const Default = Template.bind({});
Default.args = {
  title: 'Accordion Title',
  description: 'This is the description content of the accordion',
  height: 'auto',
  width: '100%',
};
