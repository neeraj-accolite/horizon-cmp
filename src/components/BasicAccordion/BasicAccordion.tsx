import { Accordion, AccordionItem } from '@nextui-org/react';
import { BasicAccordionProps } from './BasicAccordion.model';
import styles from './BasicAccordion.module.scss';

export const BasicAccordion: React.FC<BasicAccordionProps> = ({
  title,
  description,
  height,
  width,
}) => {
  return (
    <div className={styles.accordion} style={{ height, width }}>
      <Accordion>
        <AccordionItem
          key="1"
          aria-label={title}
          title={<span className={styles.boldTitle}>{title}</span>}
        >
          {description}
        </AccordionItem>
      </Accordion>
    </div>
  );
};
