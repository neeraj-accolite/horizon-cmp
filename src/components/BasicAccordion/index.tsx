export { BasicAccordion } from './BasicAccordion';
export type { BasicAccordionProps } from './BasicAccordion.model';
