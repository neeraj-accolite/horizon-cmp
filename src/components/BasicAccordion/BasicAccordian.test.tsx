import React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { BasicAccordion } from './BasicAccordion';
import { BasicAccordionProps } from './BasicAccordion.model';

const defaultProps: BasicAccordionProps = {
  title: 'Accordion Title',
  description: 'This is the description content of the accordion.',
  height: 'auto',
  width: '100%',
};

describe('UI Component - BasicAccordion', () => {
  test('renders the accordion component with default props', async () => {
    const container = render(<BasicAccordion {...defaultProps} />);
    const titleElement = await container.findByText('Accordion Title');
    expect(titleElement).toBeInTheDocument();
  });

  test('applies custom height and width styles', async () => {
    const customProps: BasicAccordionProps = {
      ...defaultProps,
      height: 'auto',
      width: '50%',
    };
    const container = render(<BasicAccordion {...customProps} />);
    const accordionElement =
      await container.findByText('Accordion Title').parentElement;

    // Check if the element exists
    expect(accordionElement).not.toBeNull();
    if (accordionElement) {
      expect(accordionElement).toHaveStyle('height: auto');
      expect(accordionElement).toHaveStyle('width: 50%');
    }
  });

  test('renders the title in bold', async () => {
    const container = render(<BasicAccordion {...defaultProps} />);
    const titleElement = await container.findByText('Accordion Title');
    expect(titleElement).toHaveClass('boldTitle');
  });
  test('matches the snapshot', () => {
    const container = render(<BasicAccordion {...defaultProps} />);
    expect(container.asFragment()).toMatchSnapshot();
  });
});
