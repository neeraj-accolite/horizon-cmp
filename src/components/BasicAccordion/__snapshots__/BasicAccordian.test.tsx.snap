// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI Component - BasicAccordion matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="accordion"
    style="height: auto; width: 100%;"
  >
    <div
      class="px-2 w-full"
      data-orientation="vertical"
    >
      <div
        aria-label="Accordion Title"
      >
        <h2>
          <button
            aria-expanded="false"
            class="flex py-4 w-full h-full gap-3 items-center tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 transition-opacity"
            id=":r6:"
            type="button"
          >
            <div
              class="flex-1 flex flex-col text-start"
            >
              <span
                class="text-foreground text-medium"
              >
                <span
                  class="boldTitle"
                >
                  Accordion Title
                </span>
              </span>
            </div>
            <span
              aria-hidden="true"
              class="text-default-400 transition-transform rotate-0 data-[open=true]:-rotate-90 rtl:-rotate-180 rtl:data-[open=true]:-rotate-90"
            >
              <svg
                aria-hidden="true"
                fill="none"
                focusable="false"
                height="1em"
                role="presentation"
                viewBox="0 0 24 24"
                width="1em"
              >
                <path
                  d="M15.5 19l-7-7 7-7"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.5"
                />
              </svg>
            </span>
          </button>
        </h2>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
